client/src/components/App.jsx

```diff
@@ -64,17 +64,31 @@ class App extends React.Component {
     this.setState({
       rating: roundedRating,
     });
+    if (cb) {
+      cb(roundedRating);
+    }
   };
 
-  getMetadata = () => {
-    const { currentItem } = this.state;
-    axios.get(`/api/reviews/meta/${currentItem.id}`)
-      .then((res) => {
-        this.calcAvgRating(res.data.ratings);
-      })
-      .catch((err) => {
-        console.log(err);
-      });
+  getMetadata = (id, cb) => {
+    console.log('cb', cb);
+    if (!id) {
+      const { currentItem } = this.state;
+      axios.get(`/api/reviews/meta/${currentItem.id}`)
+        .then((res) => {
+          this.calcAvgRating(res.data.ratings, cb);
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    } else {
+      axios.get(`/api/reviews/meta/${id}`)
+        .then((res) => {
+          this.calcAvgRating(res.data.ratings, cb);
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    }
```

Comment:
Looks good, just one small nitpick: Let's make it more DRY by setting a variable for the item's id and just inserting it and then do the rest of the axios.get stuff.

Something like: 

let itemId;

if (id) {
  itemId = id
} else {
  itemId = currentItem.id
}

and then just inserting that id into the endpoint